Evo stone fix

fuws opened the pull request:
The evolution stone menu had three issues:

1. Major issue: it's showing all base form Pokemon that can evolve using said stone, including those that the player doesn't have.
2. Major issue: Eevee evolutions were not coded correctly.
3. It doesn't have any indicator for if the player already has the evolved form (caught or shiny).

What this change looks like:

I do not have Exeggcute, so it doesn't show up for Grass Stone:

![image](https://user-images.githubusercontent.com/36806183/54797297-e6bc9700-4c29-11e9-9762-6004bf80ba6f.png)


Correctly shows which evolved form I have and not have, and whether I have them shiny. Eevee evolutions also appear correctly now.

![image](https://user-images.githubusercontent.com/36806183/54797323-08b61980-4c2a-11e9-8da9-d740068f1e8c.png)

If you look closely, you'll see that the regular pokeball is a bit farther from the right than the shiny pokeball. This is something I was not successful at addressing. I feel it's a minor issue that can be put into backlog.

Ishadijcks commented:
Looks great!

Do you know what is causing the pokeball difference? Might be an issue in the sprite?

fuws commented:
I currently use two different `<span>`s with `data-bind` conditional on caught status. Maybe the span order is causing this. I tried to put `data-bind` at the `<img>` level and it didn't work. I can look into that more.

fuws commented:
I fixed the pokeball appearance difference. I just moved the padding up one level.

New look:
![image](https://user-images.githubusercontent.com/36806183/54888878-8f136b00-4e77-11e9-86c2-5adecc0d147f.png)

Ishadijcks commented on src/scripts/items/EvolutionStone.ts:
```diff
@@ -14,33 +14,40 @@ class EvolutionStone extends Item {
         else {
-            let evolution: string = PokemonHelper.getPokemonByName(pokemon).evolution;
-            player.capturePokemon(evolution, shiny, false);
+            return PokemonHelper.getPokemonByName(pokemon).evolution;
```
Maybe build a check in case this doesn't return anything (i.e. we forgot to hardcode a special evolution).

Ishadijcks commented on src/scripts/items/EvolutionStone.ts:
```diff
@@ -14,33 +14,40 @@ class EvolutionStone extends Item {
-        if(pokemon == "Eevee") {
-            switch (this.type) {
+        let evolution = EvolutionStone.computeEvolution(this.type, pokemon);
+        player.capturePokemon(evolution, shiny, false);
```
Or build the check here, see other comment

fuws replied:
I don't think we ever run into that problem, because if a stone is coded incorrectly, then `PokemonHelper.getPokemonsWithEvolution` would see that a pokemon doesn't evolve with this stone, and wouldn't show this pokemon on the UI. So the player has no way of selecting it. 

Another way of saying this is, players are only able to select the pokemons that we have coded correctly. So an incorrect coding won't result in page rendering failure; it'll result in the base form pokemon not showing.

Ishadijcks replied:
Good point

Ishadijcks commented on src/scripts/items/EvolutionStone.ts:
```diff
@@ -14,33 +14,40 @@ class EvolutionStone extends Item {
-            switch (this.type) {
+        let evolution = EvolutionStone.computeEvolution(this.type, pokemon);
+        player.capturePokemon(evolution, shiny, false);
+        return shiny;
```
Why return the boolean `shiny`?

fuws replied:
It returns `shiny` because `ItemHandler.useStones` needs it to determine when to stop using the stones if player selects "Until Shiny". But the implementation of `ItemHandler.useStones` is a bit flawed, which I can fix in a separate PR.

fuws commented:
I updated `computeEvolution` so we don't have to hard code the Eevee and Gloom evolutions in code here. It assumes the ordering of stones and evolved forms are consistent (which they are). I checked that the page renders correctly.

RedSparr0w commented on src/scripts/pokemons/PokemonList.ts:
```diff
@@ -1795,7 +1795,7 @@ const pokemonList = [
         "catchRate": 45,
         "evolution": "Vaporeon, Jolteon, Flareon, Espeon, Umbreon",
-        "evoLevel": "Water Stone, Thunder Stone, Fire Stone, Time Stone, Time Stone",
+        "evoLevel": "Water_stone, Thunder_stone, Fire_stone, Time Stone, Time Stone",
```
For consistency:
```suggestion
        "evoLevel": "Water_stone, Thunder_stone, Fire_stone, Time_Stone, Time_Stone",
```

RedSparr0w replied:
Also on a side note, Would it be better to store these as an Array:
```javascript
"evoLevel": ["Water_stone", "Thunder_stone", "Fire_stone", "Time_Stone", "Time_Stone"],
```

Or even something like this:
```javascript
"evoLevel": {
  "Water_stone": ["Vaporeon"],
  "Thunder_stone": ["Jolteon"],
  "Fire_stone": ["Flareon"],
  "Time_Stone": ["Espeon", "Umbreon"],
},
```

RedSparr0w commented:
> I agree with @RedSparr0w, it would be cleaner to store the 'evoLevel' in a dictionary so there is no need for string parsing. Other than that I believe this is ready to merge!

Probably fine to merge as is, then sort that out in a later PR?

Ishadijcks commented:
Fair enough, could you create an issue for it?
